### PR TITLE
Bulk of Front50 endpoint authorization checks

### PR DESCRIPTION
**Pull request by ttomsu**
This PR is based off of #123 , so please review the **second** commit only.

Adds Pipeline, Strategy, and Snapshot annotation authorization checks.  There are two endpoints I'm not sure how to secure, both of them are DELETE by [pipeline|strategy] ID. It's not immediately clear if this endpoint is even used, instead using the application/pipelineID deletion endpoint. 

@ajordens @cfieber @duftler @jtk54 PTAL.

### CODE REVIEW

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/StrategyController.groovy [hunk 1]
```diff
@@ -69,23 +78,27 @@ class StrategyController {
         pipelineStrategyDAO.create(strategy.getId(), strategy)
     }
 
+    @PreFilter("hasPermission(#filterTarget.application, 'APPLICATION', 'WRITE')")
```
Comment (cfieber) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/StrategyController.groovy]: same comment as on #123 - think a preauthorize vs prefilter

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/SnapshotsController.groovy [hunk 2]
```diff
@@ -39,17 +44,20 @@ class SnapshotsController {
...
         return snapshotDAO.getHistory(id, limit)
     }
 
+    @PostAuthorize("hasPermission(returnObject.application, 'APPLICATION', 'READ')")
     @RequestMapping(value = "/{id:.+}", method = RequestMethod.GET)
     Snapshot getCurrent(@PathVariable String id) {
         return snapshotDAO.findById(id)
     }
 
+    @PreAuthorize("hasPermission(#snapshot.application, 'APPLICATION', 'READ')")
```
Comment (jtk54) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/SnapshotsController.groovy]: This is a `'READ'` on a POST, is this what you want here?

**Comment by jtk54** [comment]
LGTM after comment.

**Comment by cfieber** [comment]
LGTM  aside from the prefilter vs preauthorize question

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/SnapshotsController.groovy (hunk 2 above)
Comment (ttomsu) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/SnapshotsController.groovy]: Nice catch!

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/StrategyController.groovy (hunk 1 above)
Comment (ttomsu) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/StrategyController.groovy]: Commented out. Will address in the next PR.

**Pull request merged**